synology_dsm/api/storage/storage.py

```diff
@@ -0,0 +1,156 @@
+        volume = self._get_volume(volume_id)
+        return volume.get("device_type")
+
+    def volume_size_total(self, volume_id, human_readable=True):
+        """Total size of volume."""
+        volume = self._get_volume(volume_id)
+        if volume.get("size"):
+            return_data = int(volume["size"]["total"])
+            if human_readable:
+                return SynoFormatHelper.bytes_to_readable(return_data)
+            return return_data
+        return None
+
+    def volume_size_used(self, volume_id, human_readable=True):
+        """Total used size in volume."""
+        volume = self._get_volume(volume_id)
+        if volume.get("size"):
+            return_data = int(volume["size"]["used"])
+            if human_readable:
+                return SynoFormatHelper.bytes_to_readable(return_data)
+            return return_data
+        return None
+
+    def volume_percentage_used(self, volume_id):
+        """Total used size in percentage for volume."""
+        volume = self._get_volume(volume_id)
+        if volume.get("size"):
+            total = int(volume["size"]["total"])
+            used = int(volume["size"]["used"])
+
+            if used and used > 0 and total and total > 0:
+                return round((float(used) / float(total)) * 100.0, 1)
+        return None
+
+    def volume_disk_temp_avg(self, volume_id):
+        """Average temperature of all disks making up the volume."""
+        volume = self._get_volume(volume_id)
+        if volume:
+            vol_disks = self.disks
+            if vol_disks:
+                total_temp = 0
+                total_disks = 0
+
+                for vol_disk in vol_disks:
+                    disk_temp = self.disk_temp(vol_disk)
+                    if disk_temp:
+                        total_disks += 1
+                        total_temp += disk_temp
+
+                if total_temp > 0 and total_disks > 0:
+                    return round(total_temp / total_disks, 0)
+        return None
+
+    def volume_disk_temp_max(self, volume_id):
+        """Maximum temperature of all disks making up the volume."""
+        volume = self._get_volume(volume_id)
+        if volume:
+            vol_disks = self.disks
+            if vol_disks:
+                max_temp = 0
+
+                for vol_disk in vol_disks:
+                    disk_temp = self.disk_temp(vol_disk)
+                    if disk_temp and disk_temp > max_temp:
+                        max_temp = disk_temp
+                return max_temp
+        return None
+
+    @property
+    def disks(self):
+        """Returns all available (internal) disks."""
+        disks = []
+        for disk in self._data.get("disks"):
+            disks.append(disk["id"])
+        return disks
+
+    def _get_disk(self, disk_id):
+        """Returns a specific disk."""
+        for disk in self._data.get("disks"):
+            if disk["id"] == disk_id:
+                return disk
+        return {}

```

Comment:
Maybe return `None` here, getting an error when doing `.get()` on this may be wanted.